Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -601,6 +614,22 @@ void WiiSocket::Update(bool read, bool write, bool except)
   }
 }
 
+bool WiiSocket::HasTimeout()
```

Comment:
The naming of this function bothers me a little bit, as the function doesn't just check whether something has timed out, but also starts a timer.